File: electron/utils/paths.ts
import os from "node:os";
import path from "node:path";

export function getSpicetifyxDir(): string {
  return path.join(os.homedir(), ".spicetifyx");
}

export function getSpicetifyExec(): string {
  const dir = getSpicetifyxDir();
  return os.platform() === "win32"
    ? path.join(dir, "spicetify.exe")
    : path.join(dir, "spicetify");
}

export function getSpicetifyConfigDir(): string {
  return os.platform() === "win32"
    ? path.join(process.env.APPDATA!, "Spicetify")
    : path.join(os.homedir(), ".config", "spicetify");
}

export function getConfigFilePath(): string {
  return path.join(getSpicetifyConfigDir(), "config-xpui.ini");
}

export function getExtensionsDir(): string {
  return path.join(getSpicetifyConfigDir(), "Extensions");
}

export function getCustomAppsDir(): string {
  return path.join(getSpicetifyConfigDir(), "CustomApps");
}

export function getThemesDir(): string {
  return path.join(getSpicetifyConfigDir(), "Themes");
}

File: internal/helpers/paths.go
package helpers

import (
	"os"
	"path/filepath"
	"runtime"
)

func GetSpicetifyxDir() string {
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".spicetifyx")
}

func GetSpicetifyExec() string {
	dir := GetSpicetifyxDir()
	if runtime.GOOS == "windows" {
		return filepath.Join(dir, "spicetify.exe")
	}
	return filepath.Join(dir, "spicetify")
}

func GetSpicetifyConfigDir() string {
	if runtime.GOOS == "windows" {
		return filepath.Join(os.Getenv("APPDATA"), "Spicetify")
	}
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".config", "spicetify")
}

func GetConfigFilePath() string {
	return filepath.Join(GetSpicetifyConfigDir(), "config-xpui.ini")
}

func GetExtensionsDir() string {
	return filepath.Join(GetSpicetifyConfigDir(), "Extensions")
}

func GetCustomAppsDir() string {
	return filepath.Join(GetSpicetifyConfigDir(), "CustomApps")
}

func GetThemesDir() string {
	return filepath.Join(GetSpicetifyConfigDir(), "Themes")
}

func GetSettingsPath() string {
	return filepath.Join(GetSpicetifyxDir(), "settings.json")
}

func GetAppPath() string {
	return GetSpicetifyConfigDir()
}
